File: src/main/java/tool/ToolManager.java
package tool;

import core.gui.HOMainFrame;
import core.model.HOVerwaltung;
import core.model.TranslationFacility;
import tool.arenasizer.ArenaSizerDialog;
import tool.export.CsvPlayerExport;
import tool.hrfExplorer.HrfExplorerDialog;
import tool.injury.InjuryDialog;
import tool.keepertool.KeeperToolDialog;
import tool.notepad.NotepadDialog;

import javax.swing.*;


public class ToolManager {
    private final JMenu m_jmToolsMenu = new JMenu(TranslationFacility.tr("ls.menu.tools"));
    private final JMenuItem m_jmiInjuryCalculator = new JMenuItem(TranslationFacility.tr("InjuryCalculator"));
    private final JMenuItem m_jmiKeeperTool = new JMenuItem(TranslationFacility.tr("KeeperTool"));
    private final JMenuItem m_jmiNotepad = new JMenuItem(TranslationFacility.tr("Notizen"));
    private final JMenuItem m_jmiCsvPlayerExporter = new JMenuItem(TranslationFacility.tr("CSVExporter"));
    private final JMenuItem m_jmiArenaSizer = new JMenuItem(TranslationFacility.tr("ArenaSizer"));
    private final JMenuItem m_jmiHrfExplorer = new JMenuItem(TranslationFacility.tr("Tab_HRF-Explorer"));

    public JMenu getToolMenu() {
        m_jmiArenaSizer.addActionListener(e -> new ArenaSizerDialog(HOMainFrame.instance()).setVisible(true));
        m_jmToolsMenu.add(m_jmiArenaSizer);

        m_jmiKeeperTool.addActionListener(e -> new KeeperToolDialog(HOMainFrame.instance()).setVisible(true));
        m_jmToolsMenu.add(m_jmiKeeperTool);

        m_jmiInjuryCalculator.addActionListener(e -> new InjuryDialog(HOMainFrame.instance()).setVisible(true));
        m_jmToolsMenu.add(m_jmiInjuryCalculator);

        m_jmiCsvPlayerExporter.addActionListener(e -> {
            CsvPlayerExport csvExporter = new CsvPlayerExport();
            csvExporter.showSaveDialog();
        });
        m_jmToolsMenu.add(m_jmiCsvPlayerExporter);

        m_jmiNotepad.addActionListener(e -> {
            NotepadDialog notepad = new NotepadDialog(HOMainFrame.instance(), TranslationFacility.tr("Notizen"));
            notepad.setVisible(true);
        });
        m_jmToolsMenu.add(m_jmiNotepad);

        m_jmiHrfExplorer.addActionListener(e -> new HrfExplorerDialog(HOMainFrame.instance()).setVisible(true));
        m_jmToolsMenu.add(m_jmiHrfExplorer);

        return m_jmToolsMenu;
    }
}

File: src/main/java/tool/ToolManager.kt
package tool

import core.gui.HOMainFrame
import core.model.HOVerwaltung
import tool.arenasizer.ArenaSizerDialog
import tool.export.CsvPlayerExport
import tool.export.XMLExporter
import tool.hrfExplorer.HrfExplorerDialog
import tool.injury.InjuryDialog
import tool.keepertool.KeeperToolDialog
import tool.notepad.NotepadDialog
import javax.swing.JMenu
import javax.swing.JMenuItem

class ToolManager {
    val toolMenu: JMenu
        get() {
            val hoAdmin = HOVerwaltung.instance()

            val jmToolsMenu = JMenu(hoAdmin.getLanguageString("ls.menu.tools"))
            val jmiArenaSizer = JMenuItem(hoAdmin.getLanguageString("ArenaSizer"))
            jmiArenaSizer.addActionListener { _ -> ArenaSizerDialog(HOMainFrame).isVisible = true }
            jmToolsMenu.add(jmiArenaSizer)

            val jmiKeeperTool = JMenuItem(hoAdmin.getLanguageString("KeeperTool"))
            jmiKeeperTool.addActionListener { _ -> KeeperToolDialog(HOMainFrame).isVisible = true }
            jmToolsMenu.add(jmiKeeperTool)

            val jmiInjuryCalculator = JMenuItem(hoAdmin.getLanguageString("InjuryCalculator"))
            jmiInjuryCalculator.addActionListener { _ -> InjuryDialog(HOMainFrame).isVisible = true }
            jmToolsMenu.add(jmiInjuryCalculator)

            val jmiExporter = JMenuItem(hoAdmin.getLanguageString("XMLExporter"))
            jmiExporter.addActionListener { _ ->
                val exporter = XMLExporter()
                exporter.doExport()
            }
            jmToolsMenu.add(jmiExporter)

            val jmiCsvPlayerExporter = JMenuItem(hoAdmin.getLanguageString("CSVExporter"))
            jmiCsvPlayerExporter.addActionListener { _ ->
                val csvExporter = CsvPlayerExport()
                csvExporter.showSaveDialog()
            }
            jmToolsMenu.add(jmiCsvPlayerExporter)

            val jmiNotepad = JMenuItem(hoAdmin.getLanguageString("Notizen"))
            jmiNotepad.addActionListener { _ ->
                val notepad = NotepadDialog(HOMainFrame, hoAdmin.getLanguageString("Notizen"))
                notepad.isVisible = true
            }
            jmToolsMenu.add(jmiNotepad)

            val jmiHrfExplorer = JMenuItem(hoAdmin.getLanguageString("Tab_HRF-Explorer"))
            jmiHrfExplorer.addActionListener { _ -> HrfExplorerDialog(HOMainFrame).isVisible = true }
            jmToolsMenu.add(jmiHrfExplorer)

            return jmToolsMenu
        }
}
